indico/modules/events/contributions/client/js/index.jsx

```diff
@@ -29,30 +30,41 @@ import PublicationSwitch from './PublicationSwitch';
     if (!editableSubmissionButton) {
       return;
     }
-
     const availableTypes = JSON.parse(editableSubmissionButton.dataset.availableTypes);
     const {eventId, contributionId, contributionCode} = editableSubmissionButton.dataset;
 
-    let responses;
+    let responses, lastRevFiles;
     try {
-      responses = await Promise.all(
-        availableTypes.map(type => indicoAxios.get(fileTypesURL({confId: eventId, type})))
-      );
+      responses = await Promise.all([
+        Promise.all(
+          availableTypes.map(type => indicoAxios.get(fileTypesURL({confId: eventId, type})))
+        ),
+        indicoAxios.get(paperInfoURL({confId: eventId, contrib_id: contributionId}), {
+          validateStatus: status => (status >= 200 && status < 300) || status === 404,
+        }),
+      ]);
     } catch (e) {
       handleAxiosError(e);
       return;
     }
-
     const fileTypes = _.fromPairs(
-      responses.map((response, index) => [availableTypes[index], camelizeKeys(response.data)])
+      responses[0].map((response, index) => [availableTypes[index], camelizeKeys(response.data)])
     );
 
+    if (responses[1]) {
+      const {isInFinalState, lastRevision} = camelizeKeys(responses[1].data);
+      if (isInFinalState && lastRevision) {
+        lastRevFiles = lastRevision.files;
+      }
+    }
+
     ReactDOM.render(
       <EditableSubmissionButton
         fileTypes={fileTypes}
-        eventId={eventId}
-        contributionId={contributionId}
+        eventId={+eventId}
```

Comment:
what's this `+` for? 

Sorry if it has already been discussed, just point me to that.

Reply:
Returns the numerical representation, similar to `Number()`
https://github.com/indico/indico/pull/4489#discussion_r442167578